src/passes/transform/cse.cpp

```diff
@@ -0,0 +1,174 @@
+#include "coreir/passes/transform/cse.h"
```

Comment:
It took me a while to gather that cse => common subexpression elimination. I also see that it is part of the LLVM lexicon however.  A comment either here or in the header, that this is a common subexpression elimination pass would be useful. And exactly what types of subexpressions it can do would be useful for those who want to use the pass.